asset-transfer-basic/application-java/src/main/java/application/java/ClientUtilities.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package application.java;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.hyperledger.fabric.gateway.Contract;
+import org.hyperledger.fabric.gateway.Gateway;
+import org.hyperledger.fabric.gateway.Network;
+import org.hyperledger.fabric.gateway.Wallet;
+import org.hyperledger.fabric.gateway.Wallets;
+
+public class ClientUtilities 
+{
+	static {
+		System.setProperty("org.hyperledger.fabric.sdk.service_discovery.as_localhost", "true");
+	}
+
+    // helper function for getting connected to the gateway
+	public static Gateway connect() throws Exception{
+		// Load a file system based wallet for managing identities.
+		Path walletPath = Paths.get("wallet");
+		Wallet wallet = Wallets.newFileSystemWallet(walletPath);
+		// load a CCP
+		Path networkConfigPath = Paths.get("..", "..", "test-network", "organizations", "peerOrganizations", "org1.example.com", "connection-org1.yaml");
+
+		Gateway.Builder builder = Gateway.createBuilder();
+		builder.identity(wallet, "appUser").networkConfig(networkConfigPath).discovery(true);
+		return builder.connect();
+	}
+
+	// connects to the chaincode and calls InitLedger()
+	public static void InitLedger() throws Exception {
+
+		// create a gateway connection
+		try (Gateway gateway = connect()) {
+
+			// get the network and contract
+			Network network = gateway.getNetwork("mychannel");
+			Contract contract = network.getContract("basic");
+
+			byte[] result;
+
+			contract.submitTransaction("InitLedger");
+			System.out.println("Ledger has been initialized with 6 assets");
+			gateway.close();
+		}
+		catch(Exception e){
+			System.out.println("Error connecting to gateway: " + e);
+		}
+	}
+
+	// connects to the chaincode and calls CreateAsset()
+	public static void CreateAsset(final String assetID, final String color, final String size,
+        final String owner, final String appraisedValue) throws Exception {
+
+		// create a gateway connection
+		try (Gateway gateway = connect()) {
```

Comment:
Call connect only once in a sample transaction sequence. see my comment in the unit test file

calling connect() is expensive, since it reads wallet & config files possibly.
Avoid calling it  in each txn util  like CreateAsset , ReadAsset

Reply:
Good point, I restructured the app so that connect() is only called once.